webapp/channels/src/packages/mattermost-redux/src/selectors/entities/channel_categories.ts

```diff
@@ -447,13 +447,18 @@ export function makeGetChannelsByCategory() {
 
         const channelsByCategory: RelationOneToOne<ChannelCategory, Channel[]> = {};
 
+        // TODO: This avoids some rendering, but there is a bigger issue underneath
+        // Every time myPreferences or myChannels change (which can happen for many
```

Comment:
this is a really good point here the Preference is another reason we need to make it performant